### PR TITLE
Update infinite_sheds.py to add shaded fraction to returned variables in infinite_sheds.get_irradiance and infinite_sheds.get_irradiance_poa

### PR DESCRIPTION
**Pull request by williamhobbs**
Added shaded fraction to returned variables in `infinite_sheds.get_irradiance` and `infinite_sheds.get_irradiance_poa`. 

Related discussion is here https://github.com/pvlib/pvlib-python/discussions/1869. 

 - [x] Closes https://github.com/pvlib/pvlib-python/issues/1870
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] ~Updates entries in [`docs/sphinx/source/reference`](https://github.com/pvlib/pvlib-python/blob/main/docs/sphinx/source/reference) for API changes.~
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/main/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels (including `remote-data`) and Milestone are assigned to the Pull Request and linked Issue.

I have relatively little experience with PRs and the checklist above, so I'll likely need some guidance.

### CODE REVIEW

**Comment by kandersolar** [comment]
Thanks @williamhobbs, these code changes seem right to me.  This PR will need some testing in `pvlib/tests/bifacial/test_infinite_sheds.py` to verify that the new columns exist and have the expected values, and a note in the v0.10.3 what's new file (you may have to update your branch with changes from "upstream" for this file to exist).

A side note about procedure: you opened this PR based on changes to your `main` branch, which isn't necessarily a problem here, but comes with the downside of not being able to open more than one PR (a single branch can only have one set of changes), and adding commits to your `main` means it has a different history from pvlib's `main`, which might be a headache for you in the future.  In general it is better to start a new branch for each PR.  Since this one is already opened, it makes sense to just continue on with this one.  But after this PR is merged, you might want to re-fork pvlib and start fresh to get a clean `main` branch again.

**Comment by williamhobbs** [comment]
@kandersolar, thanks for the tip on branches. I was actually wondering about that when I created a branch for this PR...

I've updated the what's new file. 

A question on testing: Does this mean that in this part of `test_get_irradiance_poa`,

https://github.com/williamhobbs/pvlib-python/blob/192a556bade8d18c2758946a8c1218bef6722906/pvlib/tests/bifacial/test_infinite_sheds.py#L96-L105

I need to add something like,

```python
expected_shaded_fraction = np.array([0.2]) # 0.2 is a placeholder, need to add the right value
assert np.isclose(res['shaded_fraction'], expected_shaded_fraction)
```

and then do something similar further down for `test_get_irradiance`?

**Comment by kandersolar** [comment]
Yep, that's the idea.  Might as well add similar checks in the array/Series sections too.

**Comment by williamhobbs** [comment]
Ok, I added tests to `pvlib/tests/bifacial/test_infinite_sheds.py`.

For all of the combinations of solar position and array geometry that were included in the testing, shaded fraction is always zero. Is that ok? I could change, e.g., `solar_zenith` and `surface_tilt` to get a non-zero shaded fraction, but that would also change expected values for most/all other outputs. I was also thinking I could add a new test function like `test_shaded_fraction_in_get_irradiance` that only checks shaded_fraction in a case with shade, but I'm not sure if that's the right way to setup tests...

**Comment by kandersolar** [comment]
>  I was also thinking I could add a new test function like test_shaded_fraction_in_get_irradiance that only checks shaded_fraction in a case with shade, but I'm not sure if that's the right way to setup tests...

That would be fine.  In this case I think it's okay to not add that test since the functionality of the shaded fraction calculation is already tested elsewhere (e.g. `test__shaded_fraction_array`).  Tests for this PR are more to ensure that the calculation results are being returned as expected, less about the actual values themselves.

**Comment by williamhobbs** [comment]
After fixing a copy-paste mistake, it looks like I got the testing working. Can I check the boxes above for updating entries for API changes (which I don't think is relevant here?), documentation, and ready for detailed review?

**Comment by kandersolar** [comment]
Yep!  You can also ~strikethrough~ items that don't apply by wrapping in ~ like this: `~Updates entries...~`.

**Review by kandersolar — APPROVED** [review]
LGTM!

My only comment is that someone not familiar with the model might be surprised that `shaded_fraction_back` is zero most of the time.  Perhaps the docs should specify that this column represents specifically row-to-row shade, and not "sun is behind the POA" shade?  Fine with me to leave as is though.

**Comment by williamhobbs** [comment]
> My only comment is that someone not familiar with the model might be surprised that `shaded_fraction_back` is zero most of the time. Perhaps the docs should specify that this column represents specifically row-to-row shade, and not "sun is behind the POA" shade? Fine with me to leave as is though.

I'd be fine with that, but I might defer to you, @kandersolar, or @mikofski about how to describe that clearly.

File: pvlib/bifacial/infinite_sheds.py
Comment (kandersolar) [inline pvlib/bifacial/infinite_sheds.py:509]:
```suggestion
    - ``shaded_fraction_front`` : fraction of the front surface,
      going up from the bottom, affected by row-to-row beam shading.
      [unitless]
```

Maybe this?  And likewise for `shaded_fraction_back`

**Review by kandersolar — COMMENTED** [review]
One idea for parameter description wording below. 

Also @williamhobbs there's a merge conflict now that `pvlib:main` has been updated by other PRs in the time since this one was opened.  You can resolve the conflict locally using `git` commands and your text editor or using GitHub's tool at the bottom of this PR page.

**Comment by williamhobbs** [comment]
Thanks, @kandersolar. I think I've updated/fixed everything.

File: pvlib/bifacial/infinite_sheds.py
Comment (cwhanse) [inline pvlib/bifacial/infinite_sheds.py:279]: I think this description (and for the corresponding front/back quantities) should be similar to that used for `f_x` in `_shaded_fraction`:

"Fraction of row slant height from the bottom that is shaded from direct irradiance by adjacent rows."

Comment (williamhobbs) [reply pvlib/bifacial/infinite_sheds.py:279]: @cwhanse  - just to be clear, you are suggesting I revert back to the way it was, removing the changes that @kandersolar suggested?

I.e., revert back to,

in `get_irradiance_poa`:
```
    - ``shaded_fraction`` : fraction of row slant height from the bottom that
      is shaded from direct irradiance. [unitless]
```

in `get_irradiance`:
```
    - ``shaded_fraction_front`` : fraction of row slant height from the
      bottom that is shaded from direct irradiance on the front surface.
      [unitless]
.
.
.
    - ``shaded_fraction_back`` : fraction of row slant height from the
      bottom that is shaded from direct irradiance on the back surface.
      [unitless]
```

Comment (kandersolar) [reply pvlib/bifacial/infinite_sheds.py:279]: That's closer to what @williamhobbs originally had before I tried to improve it :) so long as it mentions something about the shading coming from other rows, I am fine with anything here.

**Review by cwhanse — APPROVED** [review]

**Review by mikofski — APPROVED** [review]
Is this @williamhobbs  first PR? Coincidence that it’s one of my favorites of all time?! Congrats Will 🎉🥳

**Comment by kandersolar** [comment]
Thanks @williamhobbs!

**Pull request merged**
